src/forms/yup.js

```diff
@@ -24,7 +24,11 @@ const ARRAY_INDEX_REGEX = /\[([^)]+)\]/;
 yup.setLocale({
   mixed: {
     default: params => ({ key: 'form.validation_field_invalid', params }),
-    required: params => ({ key: 'form.validation_field_required', params }),
+    required: params => {
+      return params.path === 'location'
```

Comment:
Issue: `yup.js` ideally should not be aware of what `location` is. I think for this you should be able to define a custom required message for the specific validation:
https://github.com/techmatters/terraso-web-client/blob/main/src/landscape/components/LandscapeForm/KeyInfoStep.js#LL41C23-L41C46
```
location: yup.string().trim().required(),
```
Maybe something like this:
```
location: yup.string().trim().required(params => ({ key: 'form.validation_select_field_required', params })),
```